backend/tests/integration/pidstore/minters/test_minters_texkey.py

```diff
@@ -252,3 +252,25 @@ def test_minter_undeleting_record(inspire_app, override_config):
         PersistentIdentifier.pid_type == "texkey",
     ).one()
     assert texkey.status == PIDStatus.REGISTERED
+
+
+def test_minter_deletes_texkey_missing_in_metadata(inspire_app, override_config):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+    generated_texkeys = record["texkeys"]
+
+    assert len(generated_texkeys) == 1
+
+    data = dict(record)
+    del data["texkeys"]
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record.update(data)
+    assert record["texkeys"] != generated_texkeys
```

Comment:
this might randomly fail (rarely) if the same texkey is assigned again. Maybe assign 2 texkeys at the beginning, then remove one and check that it is actually removed instead?

Reply:
It will be easier to just do second update record with different year in publication_info, Then it will be impossible to generate identical texkey on this update.